### PR TITLE
add dataset create and pull

### PR DESCRIPTION
**Pull request by moshe-kabala**
- Move part of config into new dataset package
- Add utils for dataset package and use it
- Add ask dataset and get dataset by id
- Add download file function
- Add extract tar and download and extract tar
- Add create and pull commands
- Update API

### CODE REVIEW

**Review by YuvalFeldman — APPROVED** [review]

File: pkg/datasets/utils.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	. "github.com/tensorleap/cli-go/pkg/api"
+	"github.com/tensorleap/cli-go/pkg/local"
+	"github.com/tensorleap/cli-go/pkg/log"
+	"github.com/tensorleap/cli-go/pkg/tensorleapapi"
+)
+
+var EmptyDatasetVersionError = fmt.Errorf("Dataset is empty")
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/datasets/utils.go:14]: This is a common pattern for comparing err in go, I use this in the next commit (if the dataset is empty I using template)

File: pkg/datasets/utils.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		Message: "Enter dataset name",
+	}
+	survey.AskOne(prompt, &name)
+	return
+}
+
+func AskForDataset(ctx context.Context) (*tensorleapapi.Dataset, error) {
+	data, _, err := ApiClient.GetDatasets(ctx).Execute()
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/datasets/utils.go:27]: I use this function on pull command (ask the user which dataset to pull)

File: pkg/datasets/config.go [hunk 3]
```diff
@@ -17,21 +18,40 @@ type DatasetConfig struct {
 	IncludePatterns []string `yaml:"include"`
 }
 
+func NewDatasetConfig(datasetId string, entryFile string, files []string) *DatasetConfig {
+	if len(entryFile) == 0 {
+		entryFile = "tensorleap.py"
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/datasets/config.go:24]: Will check it

File: pkg/datasets/utils.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/datasets/utils.go:14]: It can't be const when calling a function

**Pull request merged**